cc-snapshot

```diff
@@ -191,223 +269,240 @@ fi
+    run_cmd yum install -y libguestfs-xfs
+    run_cmd yum update -y lvm2 # must be at least 2.02-171 to avoid bug https://bugzilla.redhat.com/show_bug.cgi?id=1475018
+  fi
 
 
-if [ $DISTRO = $UBUNTU ]; then
-  # Ensure /boot/efi directory exists and install grub-efi if it doesn't
-  if [ ! -d /boot/efi ]; then
-    echo "Creating /boot/efi directory..."
-    sudo mkdir -p /boot/efi
-    echo "Installing grub-efi..."
-    apt-get install -yq grub-efi
+  if [ $DISTRO = $UBUNTU ]; then
+    # Ensure /boot/efi directory exists and install grub-efi if it doesn't
+    if [ ! -d /boot/efi ]; then
+      echo "Creating /boot/efi directory..."
+      run_cmd sudo mkdir -p /boot/efi
+      echo "Installing grub-efi..."
+      run_cmd apt-get install -yq grub-efi
+    fi
   fi
-fi
-
+}
 
 # Create a tar file of the contents of your instance:
-declare -a TAR_ARGS=(--selinux --acls --numeric-owner --one-file-system --exclude-from $EXCLUDE_FROM)
-if [ $DISTRO = $CENTOS ]; then
-  TAR_ARGS+=(--xattrs)
-fi
-
-# The following code is inspired from http://unix.stackexchange.com/questions/82598
-n=0
-until [ $n -ge 5 ]
-do
-  tar --create --file $CC_SNAPSHOT_TAR_PATH ${TAR_ARGS[@]} / && break
-  n=$[$n+1]
-  sleep 15
-done
+prepare_tarball(){
+  declare -a TAR_ARGS=(--selinux --acls --numeric-owner --one-file-system --exclude-from $EXCLUDE_FROM)
+  if [ $DISTRO = $CENTOS ]; then
+    TAR_ARGS+=(--xattrs)
+  fi
 
-# Get the size of tar and warn user if exceeds threshold property
-SNAPSHOT_TAR_SIZE=$(($(stat -c %s $CC_SNAPSHOT_TAR_PATH) / 1024 / 1024 ))
-LARGE_TAR_CONTINUE='yes'
-if [ $SNAPSHOT_TAR_SIZE -gt $CC_SNAPSHOT_MAX_TARBALL_SIZE ] && [ "$IGNORE_WARNING" == false ]; then
-  echo "WARNING: snapshot is too large ($SNAPSHOT_TAR_SIZE)! May cause issues."
-  echo 'Do you want to continue? (yes/no)'
-  read -r LARGE_TAR_CONTINUE_INPUT
-  export LARGE_TAR_CONTINUE=$LARGE_TAR_CONTINUE_INPUT
-fi
+  # The following code is inspired from http://unix.stackexchange.com/questions/82598
+  n=0
+  until [ $n -ge 5 ]
+  do
+    run_cmd tar --create --file $CC_SNAPSHOT_TAR_PATH ${TAR_ARGS[@]} -C "$SOURCE_DIR" . && break
+    n=$[$n+1]
+    sleep 15
+  done
+}
 
-if [ ${LARGE_TAR_CONTINUE,,} != 'yes' ]; then
-  echo 'Aborting...'
-  exit 0
-fi
+create_disk_image(){
+  if [ $DISTRO = $UBUNTU ]; then
+    if [ $UBUNTU_VERSION != "trusty" ]; then
+      run_cmd apt-get install -yq libguestfs-tools
+    fi
 
-if [ $DISTRO = $UBUNTU ]; then
-  if [ $UBUNTU_VERSION != "trusty" ]; then
-    apt-get install -yq libguestfs-tools
+    # Update guestfs appliances (prevent an error with virt-make-fs);
+    # BUG: one package cannot be found (kpartx) and cause update-guestfs-appliance
+    #      to exit with an error code;
+    run_cmd apt-get -yq update # APT repos may need an update in order to find kpartx
+    run_cmd apt-get install -yq kpartx
+    if [ "$(which update-guestfs-appliance)" != "" ]; then
+      run_cmd update-guestfs-appliance || true
+    fi
   fi
```

Comment:
is there a reason this chunk didn't get moved into install_prerequisites?

Reply:
I think probably just an oversight. We can move it at some point.